arch/arm/boot/dts/overlays/revpi-flat-s-2022-overlay.dts

```diff
@@ -0,0 +1,401 @@
+				compatible = "ti,tca6416";
+				reg = <0x20>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			rtc@51 {
+				compatible = "nxp,pcf2129";
+				reg = <0x51>;
+				reset-source;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@3 {
+		target = <&spi5>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 17 GPIO_ACTIVE_LOW>,
+				   <&gpio 18 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi_5_pins &spi_5_cs_pins>;
+			status = "okay";
+
+			tpm: tpm@0 {
+				compatible = "infineon,slb9670";
+				pinctrl-names = "default";
+				pinctrl-0 = <&tpm_pins>;
+				reg = <0>;
+				spi-max-frequency = <1000000>;
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <22 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+
+			ksz8795: switch@1 {
+				// TODO: use DSA?
+				compatible = "microchip,ksz8795";
+				// NOTE: reduce this if device is not working properly
+				spi-max-frequency = <20000000>;
+				reg = <1>;
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <44 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@4 {
+		target = <&spi6>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 6 GPIO_ACTIVE_LOW>,
+				   <&gpio 10 GPIO_ACTIVE_LOW>,
+				   <&gpio 27 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi_6_pins &spi_6_cs_pins>;
+			status = "okay";
+
+			aout: dac@0 {
+				compatible = "ti,dac7512";
+				reg = <0>;
+				spi-max-frequency = <20000000>;
+				spi-cpol;
+				vcc-supply = <&aout_vref>;
+				status = "okay";
+			};
+
+			ain: adc@1 {
+				compatible = "microchip,mcp3550-50";
+				reg = <1>;
+				spi-max-frequency = <5000000>;
+				vref-supply = <&ain_vref>;
+				status = "okay";
+			};
+
+			mbus: rf@2 {
+				compatible = "spidev", "ti,cc1101";
+				pinctrl-names = "default";
+				pinctrl-0 = <&mbus_pins>;
+				reg = <2>;
+				spi-max-frequency = <5000000>;
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <45 IRQ_TYPE_LEVEL_HIGH>;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@5 {
+		target = <&xhci>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		xhci_usb: __overlay__ {
+			status = "okay";
+
+			hub@1 {
+				/* SMSC LAN9514 */
+				compatible = "usb424,9514";
+				reg = <1>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				eth0: ethernet@1 {
+					compatible = "usb424,ec00";
+					reg = <1>;
+				};
+
+				eth1: ethernet@4 {
+					/* SMSC LAN9500, attached to KSZ8795 */
+					compatible = "usb424,9500";
+					reg = <4>;
+				};
+			};
+		};
+	};
+
+	fragment@6 {
+		target = <&sdhci>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&sdhci_pins>;
+			bus-width = <4>;
+			non-removable;
+			max-frequency = <28000000>;
+			mmc-pwrseq = <&wlan_pwrseq>;
+			status = "okay";
+		};
```

Comment:
What about a comment, that the sdxxx-wlan is attached to the SDIO, other wise it is not clear from a look at the device-tree

Reply:
I moved the node below the SDHCI node now, so the relationship should be clearer